### ISSUE TITLE
Unused mut lint doesn't warn on declare first - initialize later pattern

### ISSUE THREAD

**Issue by japaric** [rust-lang/rust#15278]
Script to reproduce:

``` rust
fn main() {
    // `a` doesn't need to be mutable
    let mut a: int;
    a = 42;
    println!("{}", a);
}
```

No warnings on the output.

``` bash
$ rustc --version
rustc 0.11.0-pre (287dcb77b34b421256c1d9cf26f3101276d2f486 2014-06-30 10:26:28 +0000)
```

**Comment by steveklabnik** [comment]
Triage: still true as of `rustc 1.3.0-nightly (8fa2185e0 2015-06-27)`

**Comment by arielb1** [comment]
Fixed in 1.3.
